indico/modules/events/registration/templates/management/_registration_details.html

```diff
@@ -179,6 +180,13 @@
                     {% trans %}Get ticket{% endtrans %}
                 </a>
             {% endif %}
+            {% if can_get_document %}
+                <a href="#" class="i-button accept icon-agreement"
+                data-callback="printReceipts"
```

Comment:
Two things:

1) the alignment of those attributes does not match the style we use everywhere else
2) currently the page is not reloaded after generating a document, so it does not show up in the list of documents

this diff should fix both:

```patch
diff --git a/indico/modules/events/registration/templates/management/_registration_details.html b/indico/modules/events/registration/templates/management/_registration_details.html
index 1bb2bc27b6..659ecdf159 100644
--- a/indico/modules/events/registration/templates/management/_registration_details.html
+++ b/indico/modules/events/registration/templates/management/_registration_details.html
@@ -182,8 +182,10 @@
             {% endif %}
             {% if can_get_document %}
                 <a href="#" class="i-button accept icon-agreement"
-                data-callback="printReceipts"
-                data-params="{{ {'registration_id': [registration.id|string], 'event_id': registration.registration_form.event.id} | tojson | forceescape }}">
+                   data-callback="printReceipts"
+                   data-params="{{ {'registration_id': [registration.id|string],
+                                    'event_id': registration.registration_form.event.id,
+                                    'reload_after': true} | tojson | forceescape }}">
                     {% trans %}Get document{% endtrans %}
                 </a>
             {% endif %}
diff --git a/indico/modules/receipts/client/js/index.js b/indico/modules/receipts/client/js/index.js
index d32eb748f6..9e0b06fe57 100644
--- a/indico/modules/receipts/client/js/index.js
+++ b/indico/modules/receipts/client/js/index.js
@@ -11,8 +11,22 @@ import {injectModal} from 'indico/react/util';

 import PrintReceiptsModal from './printing/PrintReceiptsModal';

-window.printReceipts = function({registration_id: registrationIds, event_id: eventId}) {
+window.printReceipts = function({
+  registration_id: registrationIds,
+  event_id: eventId,
+  reload_after: reloadAfter,
+}) {
   injectModal(resolve => (
-    <PrintReceiptsModal onClose={resolve} registrationIds={registrationIds} eventId={eventId} />
+    <PrintReceiptsModal
+      onClose={generated => {
+        if (generated && reloadAfter) {
+          location.reload();
+        } else {
+          resolve();
+        }
+      }}
+      registrationIds={registrationIds}
+      eventId={eventId}
+    />
   ));
 };
diff --git a/indico/modules/receipts/client/js/printing/PrintReceiptsModal.jsx b/indico/modules/receipts/client/js/printing/PrintReceiptsModal.jsx
index 1e3bd8f119..725e61fa67 100644
--- a/indico/modules/receipts/client/js/printing/PrintReceiptsModal.jsx
+++ b/indico/modules/receipts/client/js/printing/PrintReceiptsModal.jsx
@@ -141,7 +141,7 @@ export default function PrintReceiptsModal({onClose, registrationIds, eventId})
         filename: 'document',
         save_config: true,
       }}
-      onClose={onClose}
+      onClose={() => onClose(receiptIds.length > 0)}
       header={Translate.string('Generate Documents')}
       keepDirtyOnReinitialize
       noSubmitButton
```